x-pack/plugins/apm/public/components/app/settings/agent_configurations/agent_configuration_create_edit/service_page/service_page.tsx

```diff
@@ -81,6 +94,14 @@ export function ServicePage({ newConfig, setNewConfig, onClickNext }: Props) {
     })
   );
 
+  useEffect(() => {
+    setIsAgentConfigurationSupported(
+      newConfig.agent_name
+        ? !isOpenTelemetryAgentName(newConfig.agent_name as AgentName)
+        : true
+    );
+  }, [newConfig.agent_name]);
```

Comment:
Do this need to be stateful? I don't think it's problematic doing this in the render function:

```ts
const isAgentConfigurationSupported  = isOpenTelemetryAgentName(newConfig.agent_name as AgentName);
```
(if you really care about perf you could `useMemo` but I don't think that's needed)

Reply:
Ah yes, we don't need this to be stateful, i have changed the logic.
Regarding the use of `useMemo`, i don't feel it will add any added advantage as `isOpenTelemetryAgentName` is simply a `Array.includes` on an array of length close to 10, so not much of a gain.